SessionMessagingKit/Messages/Message.swift

```diff
@@ -364,13 +365,36 @@ public extension Message {
                 threadVariant: .openGroup
             )
         for (encodedEmoji, rawReaction) in reactions {
-            if let emoji = encodedEmoji.removingPercentEncoding,
+            if let decodedEmoji = encodedEmoji.removingPercentEncoding,
                rawReaction.count > 0,
                let reactors = rawReaction.reactors
             {
+                // Decide whether we need to add an extra reaction from current user
+                let pendingChanges = associatedPendingChanges
+                    .filter {
+                        if case .reaction(_, let emoji, _) = $0.metadata {
+                            return emoji == decodedEmoji
+                        }
+                        return false
+                    }
+                var shouldAddSelfReaction: Bool = rawReaction.you || reactors.contains(userPublicKey)
+                pendingChanges.forEach {
+                    if case .reaction(_, _, let action) = $0.metadata {
+                        switch action {
+                            case .react: shouldAddSelfReaction = true
+                            case .remove: shouldAddSelfReaction = false
+                        }
+                    }
+                }
```

Comment:
This can be cleaned up a bit to:
```
let shouldAddSelfReaction: Bool = (
    rawReaction.you ||
    reactors.contains(userPublicKey) ||
    associatedPendingChanges
        .contains { pendingChange in
            if case .reaction(_, let emoji, let action) = pendingChange.metadata {
                return (
                    emoji == decodedEmoji &&
                    action == .react
                )
            }
            
            return false
        }
)
```

Reply:
Will clean up this.

Reply:
Hmm... Take a closer look and the logic can't be changed like this. Because the logic needs to be when there is no pending changes, the source of truth will be `rawReaction.you || reactors.contains(userPublicKey)`, if there are pending changes, the the source of truth will be the last pending changes and what the server returns will be ignored.

Reply:
Ah sorry I missed that we were also forcing the result to `false` for the `remove` case 😞 

The main thing I'm looking to do is remove the mutability of the `shouldAddSelfReaction` variable so if a future dev modifies the code further in that function they need to explicitly make the variable mutable again (the current intent of this code is that after it is set the `shouldAddSelfReaction` shouldn't be getting updated) - it's not as succinct but the below should handle both cases

This also includes explicit sorting to deal with situations where the user adds/removes reactions and ends up with multiple pending changes for the same reaction (it'll be a bit weird while the add/remove API call hasn't returned but should handle the case were we are waiting on a response from the poller a little better)
```
let pendingChangeSelfReaction: Bool? = {
    // Find the newest 'PendingChange' entry with a matching emoji, if one exists, and
    // set the "self reaction" value based on it's action
    let maybePendingChange: PendingChange? = associatedPendingChanges
        .sorted(by: { lhs, rhs -> Bool in (lhs.seqNo ?? Int64.max) > (rhs.seqNo ?? Int64.max) })
        .first { pendingChange in
            if case .reaction(_, let emoji, _) = pendingChange.metadata {
                return emoji == decodedEmoji
            }
            
            return false
        }
    
    // If there is no pending change for this reaction then return nil
    guard
        let pendingChange: PendingChange = maybePendingChange,
        case .reaction(_, _, let action) = pendingChange.metadata
    else { return nil }

    // Otherwise add/remove accordingly
    return (action == .react)
}()
let shouldAddSelfReaction: Bool = (
    pendingChangeSelfReaction ??
    (rawReaction.you || reactors.contains(userPublicKey))
)
```

Reply:
Just for reference - when I add code snippets I don't generally run it to confirm it's working correctly, it's mostly just recommendations so always best to double check it makes sense 😜